crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs

```diff
@@ -136,6 +133,15 @@ impl PduBufferParsing<'_> for FrameBeginPdu {
             number_of_regions,
         })
     }
+}
+
+impl PduBufferParsing<'_> for FrameBeginPdu {
```

Comment:
question: What is this trait? It seems to be a legacy one.

Reply:
I don't know anything more than you about the trait, however it looks like it's defined in a `legacy` module and is still heavily used:

```
$ grep -rn 'impl PduBufferParsing'.
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:78:impl PduBufferParsing<'_> for ContextPdu {
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:138:impl PduBufferParsing<'_> for FrameBeginPdu {
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:170:impl PduBufferParsing<'_> for FrameEndPdu {
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:203:impl PduBufferParsing<'_> for RegionPdu {
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:424:impl PduBufferParsing<'_> for RfxRectangle {
crates/ironrdp-pdu/src/codecs/rfx/data_messages.rs:464:impl PduBufferParsing<'_> for Quant {
crates/ironrdp-pdu/src/codecs/rfx/header_messages.rs:36:impl PduBufferParsing<'_> for SyncPdu {
crates/ironrdp-pdu/src/codecs/rfx/header_messages.rs:80:impl PduBufferParsing<'_> for CodecVersionsPdu {
crates/ironrdp-pdu/src/codecs/rfx/header_messages.rs:130:impl PduBufferParsing<'_> for ChannelsPdu {
crates/ironrdp-pdu/src/codecs/rfx/header_messages.rs:211:impl PduBufferParsing<'_> for RfxChannel {
crates/ironrdp-pdu/src/codecs/rfx/header_messages.rs:245:impl PduBufferParsing<'_> for CodecVersion {
```